algorithms_task_1/student/include/impl.h

```diff
@@ -11,4 +11,9 @@
   * @return consecutive range
 */
 template<class FI, class Comparator = std::equal_to<typename std::iterator_traits<FI>::value_type>>
-std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {});
+std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {})
+{
+    FI it1 = std::adjacent_find(start, end, comp);
+    FI it2 = (it1 != end) ? std::find_if_not( std::next(it1), end, [&](typename std::iterator_traits<FI>::value_type elem){return comp(elem, *it1);} ) : end;
```

Comment:
Don't copy to lambda

Reply:
Please separate this expression, because currently it is unreadable

Reply:
Done